vowpalwabbit/learner.h

```diff
@@ -974,12 +974,14 @@ VW_WARNING_STATE_POP
 template <class DataT, class ExampleT, class BaseLearnerT>
 reduction_learner_builder<DataT, ExampleT, BaseLearnerT> make_reduction_learner(std::unique_ptr<DataT>&& data,
     BaseLearnerT* base, void (*learn_fn)(DataT&, BaseLearnerT&, ExampleT&),
-    void (*predict_fn)(DataT&, BaseLearnerT&, ExampleT&), const std::string& name)
+    void (*predict_fn)(DataT&, BaseLearnerT&, ExampleT&), const std::string& name,
+    bool learn_returns_prediction = false)
```

Comment:
This should not be a part of the `make_reduction_learner` call. The point of this interface is to reduce the number of arguments required.

```suggestion
    void (*predict_fn)(DataT&, BaseLearnerT&, ExampleT&), const std::string& name)
```

Line 984 should be: 
```c++
builder.set_learn_returns_prediction(false);
```